### DISCUSSION TITLE
在使用本库的同时单独建立dll库无法使用

### DISCUSSION THREAD

**Discussion by chenxuanw** [netease-im/NIM_Duilib_Framework#496] [Q&A]
我单独实现了一个网络模板，是一个dll，采用unicode字符集和MDd运行库。然而我的接口函数中若返回值或者参数出现string时，就会出现断言失败，Expression: __acrt_first_block == header
For information on how your program can cause an assertion
failure, see the Visual C++ documentation on asserts.尽管函数十分简单std::string test(std::string){string result=""; return result;}会在return时跳出断言失败，但我若是把参数和返回值都改为int则不会出现该问题。

**Comment by nmgwddj** [comment]
该问题与 duilib 框架无关。
可以给您的一些思路是：请注意 MSVC 编译器 Debug 和 Release 导出函数的 ABI 兼容情况，这也是为什么很多 SDK 提供的是纯 C 接口或虚函数 + 基础数据类型的原因。因为有些 C++ 对象不是标准布局，在不同的编译器间、不同的 build type 下会有 ABI 兼容问题。
